nucliadb/src/nucliadb/common/cluster/rebalance.py

```diff
@@ -115,9 +106,10 @@ async def move_paragraphs(
 
         while moved_paragraphs < max_paragraphs:
             # Take a random resource to move
-            try:
-                resource_id = random.choice(tuple(self.index[from_shard.id]))
-            except (KeyError, IndexError):
+            resource_id = await get_random_resource_from_shard(
+                self.context.kv_driver, self.kbid, from_shard.id
+            )
```

Comment:
Should you store here the batch of resource ids you just got to avoid too many queries to maindb?